src/radix_connect/mobile/deep_link_parsing/parser.rs

```diff
@@ -31,147 +39,437 @@ pub fn parse_mobile_connect_request(
+            Err(CommonError::InvalidEd25519PublicKeyFromString { .. })
+        ));
+    }
+
+    #[test]
+    fn parse_url_missing_session_id() {
+        let mut request = SampleRequestParams::new_from_text_vector();
+        request.session_id = None;
+
+        assert_deep_link_url_is_invalid(request);
+    }
+
+    #[test]
+    fn parse_url_missing_origin() {
+        let mut request = SampleRequestParams::new_from_text_vector();
+        request.origin = None;
+
+        assert_deep_link_url_is_invalid(request);
+    }
+
+    #[test]
+    fn parse_url_missing_interaction() {
+        let mut request = SampleRequestParams::new_from_text_vector();
+        request.request = None;
+
+        assert_deep_link_url_is_invalid(request);
+    }
+
+    #[test]
+    fn parse_url_missing_dapp_definition_address() {
+        let mut request = SampleRequestParams::new_from_text_vector();
+        request.dapp_definition_address = None;
+
+        assert_deep_link_url_is_invalid(request);
+    }
+
+    #[test]
+    fn parse_url_missing_signature() {
+        let mut request = SampleRequestParams::new_from_text_vector();
+        request.signature = None;
+
+        assert_deep_link_url_is_invalid(request);
+    }
+
+    #[test]
+    fn parse_url_missing_identity() {
+        let mut request = SampleRequestParams::new_from_text_vector();
+        request.identity_public_key = None;
+
+        assert_deep_link_url_is_invalid(request);
+    }
+
+    #[test]
+    fn parse_url_with_invalid_dapp_definition_address() {
+        let mut request = SampleRequestParams::new_from_text_vector();
+        request.dapp_definition_address = "invalid".to_string().into();
+
+        let invalid_dapp_definition_address_url = request.build_base_url();
+        let result =
+            parse_mobile_connect_request(invalid_dapp_definition_address_url);
+        assert!(matches!(
+            result,
+            Err(CommonError::FailedToDecodeAddressFromBech32 { .. })
+        ));
+    }
+
+    #[test]
+    fn parse_url_signature_validation_failed() {
+        let mut request = SampleRequestParams::new_from_text_vector();
+        request.signature = "93bc8fd33cdbd56bc1f7a9b46afc9615b5b42e9aad63227e71b02c57eb88f5f166406182afa82ebe8eb3bfc9e1388adfd60670d098751b1507584999be36c50f".to_string().into();
+
+        let wrogn_signature_url = request.build_base_url();
+        let result = parse_mobile_connect_request(wrogn_signature_url);
+        pretty_assertions::assert_eq!(
+            result,
+            Err(CommonError::RadixConnectMobileInvalidDappSignature)
+        );
+    }
+
+    fn assert_deep_link_url_is_invalid(request: SampleRequestParams) {
+        let bad_url = request.build_base_url();
+        let result = parse_mobile_connect_request(&bad_url);
+        pretty_assertions::assert_eq!(
+            result,
+            Err(CommonError::RadixConnectMobileInvalidRequestUrl {
+                bad_value: bad_url.to_owned(),
+            })
         );
     }
```

Comment:
I would add some trivial tests which asserts the constants, which are good test because they take 5 sec each to write but we will forever be protected against accidental changes to those important constants :)